src/components/atoms/DeviceVideo/DeviceVideo.tsx

```diff
@@ -0,0 +1,103 @@
+import React, { useCallback, useEffect, useRef, useState } from "react";
+import { faCircleNotch } from "@fortawesome/free-solid-svg-icons/faCircleNotch";
+import { FontAwesomeIcon } from "@fortawesome/react-fontawesome";
+
+import "./DeviceVideo.scss";
+
+export interface DeviceVideoProps {
+  facingMode?: "user" | "environment";
+  onPlay?: () => void;
+  onError?: (e: Error) => void;
+}
+
+export const DeviceVideo: React.FC<DeviceVideoProps> = ({
```

Comment:
`DeviceVideo` is a component that doesn't depend on outside services to display device camera video.